### PR TITLE
Implement _tz_convert_tzlocal_utc for de-duplication

### PR DESCRIPTION
**Pull request by jbrockmendel**
There is a bunch of code in conversion (and a bit in resolution, period, and tslib) that can be de-duplicated.  This is one part of ~5 steps in that process.

### CODE REVIEW

File: pandas/_libs/tslibs/conversion.pyx [hunk 1]
```diff
@@ -1205,7 +1187,6 @@ def is_date_array_normalized(ndarray[int64_t] stamps, tz=None):
         for i in range(n):
             # Adjust datetime64 timestamp, recompute datetimestruct
             pos = trans.searchsorted(stamps[i]) - 1
-            inf = tz._transition_info[pos]
```
Comment (jbrockmendel) [inline pandas/_libs/tslibs/conversion.pyx]: unrelated to the rest of the PR

File: pandas/_libs/tslibs/conversion.pyx [hunk 2]
```diff
@@ -608,32 +608,33 @@ cpdef inline datetime localize_pydatetime(datetime dt, object tz):
 # ----------------------------------------------------------------------
 # Timezone Conversion
 
-cdef inline int64_t tz_convert_tzlocal_to_utc(int64_t val, tzinfo tz):
+cdef inline int64_t _tz_convert_tzlocal_utc(int64_t val, tzinfo tz,
+                                            bint to_utc=True):
```
Comment (jreback) [inline pandas/_libs/tslibs/conversion.pyx]: looks good. can you add what this function does in doc-string. and mark as this is private.

**Review by jreback — CHANGES_REQUESTED** [review]
minor comments. prefer as a non-private, though if its*only* for this module then private is ok.

File: pandas/_libs/tslibs/conversion.pyx (hunk 2 above)
Comment (jbrockmendel) [reply pandas/_libs/tslibs/conversion.pyx]: Mind if I put the docsting in the follow-up?  There’s a couple more coming up in the next few hours.

**Comment by jbrockmendel** [comment]
Yes, the private function is just for this module.

**Review by jreback — APPROVED** [review]

File: pandas/_libs/tslibs/conversion.pyx (hunk 2 above)
Comment (jreback) [reply pandas/_libs/tslibs/conversion.pyx]: sure

**Pull request merged**